### PR TITLE
add nats jetstream statefulset

### PR DESCRIPTION
**Pull request by zhaojizhuang**
Part of #145

add nats jetstream statefulset using ` nats 2.3.4`

**Release Note**

```release-note
add nats jetstream statefulset using  nats 2.3.4

```

**Docs**

### CODE REVIEW

File: config/broker/README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-# NATS Streaming - simple installation
+# NATS - simple installation
+
+## 1. NATS Streaming Deprecation Notice :warning:
+> The NATS Streaming Server is being deprecated.  Critical bug fixes and security fixes will be applied until June of 2023.  NATS enabled applications requiring persistence should use [JetStream](https://docs.nats.io/jetstream/jetstream).
```
Comment (vaikas) [inline config/broker/README.md:4]: Do you really want to commit to almost 2 years of support here?

File: config/broker/config-br-default-channel-jsm.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 The Knative Authors
```
Comment (vaikas) [inline config/broker/config-br-default-channel-jsm.yaml:1]: nit, year

File: config/broker/README.md [hunk 3]
```diff
@@ -12,3 +16,16 @@
 
 For tuning NATS Streaming, see
 [here](https://github.com/nats-io/nats-streaming-server#configuring)
+
+## 2. NATS JetStream （recommend）
+
+1. For an installation of a simple NATS JetStream server, a setup is provided:
+
+   ```sbtshell
```
Comment (vaikas) [inline config/broker/README.md:24]: nit, I think for these we typically have been using just shell

File: config/broker/README.md (hunk 1 above)
Comment (zhaojizhuang) [reply config/broker/README.md:4]: No, we will just support several releases，2023 is the official promise of NATS STREAMING

**Comment by zhaojizhuang** [comment]
Done @vaikas

**Review by vaikas — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**